java/src/main/native/OrtJniUtil.c

```diff
@@ -220,225 +221,207 @@ jfloat convertHalfToFloat(uint16_t half) {
-    //printf("numDim %d\n",numDim);
-    int64_t* dimensions = (int64_t*) malloc(sizeof(int64_t)*numDim);
-    checkOrtStatus(jniEnv,api,api->GetDimensions(info, dimensions, numDim));
-    jint onnxTypeInt = convertFromONNXDataFormat(onnxType);
-
-    // Create the long array for the shape.
-    jlongArray shape = (*jniEnv)->NewLongArray(jniEnv, safecast_size_t_to_jsize(numDim));
-    (*jniEnv)->SetLongArrayRegion(jniEnv, shape, 0, safecast_size_t_to_jsize(numDim), (jlong*)dimensions);
-    // Free the dimensions array
-    free(dimensions);
-    dimensions = NULL;
-
-    // Create the ONNXTensorType enum
-    char *onnxTensorTypeClassName = "ai/onnxruntime/TensorInfo$OnnxTensorType";
-    jclass clazz = (*jniEnv)->FindClass(jniEnv, onnxTensorTypeClassName);
-    jmethodID onnxTensorTypeMapFromInt = (*jniEnv)->GetStaticMethodID(jniEnv,clazz, "mapFromInt", "(I)Lai/onnxruntime/TensorInfo$OnnxTensorType;");
-    jobject onnxTensorTypeJava = (*jniEnv)->CallStaticObjectMethod(jniEnv,clazz,onnxTensorTypeMapFromInt,onnxTypeInt);
-    //printf("ONNXTensorType class %p, methodID %p, object %p\n",clazz,onnxTensorTypeMapFromInt,onnxTensorTypeJava);
-
-    // Create the ONNXJavaType enum
-    char *javaDataTypeClassName = "ai/onnxruntime/OnnxJavaType";
-    clazz = (*jniEnv)->FindClass(jniEnv, javaDataTypeClassName);
-    jmethodID javaDataTypeMapFromONNXTensorType = (*jniEnv)->GetStaticMethodID(jniEnv,clazz, "mapFromOnnxTensorType", "(Lai/onnxruntime/TensorInfo$OnnxTensorType;)Lai/onnxruntime/OnnxJavaType;");
-    jobject javaDataType = (*jniEnv)->CallStaticObjectMethod(jniEnv,clazz,javaDataTypeMapFromONNXTensorType,onnxTensorTypeJava);
-    //printf("JavaDataType class %p, methodID %p, object %p\n",clazz,javaDataTypeMapFromONNXTensorType,javaDataType);
-
-    // Create the TensorInfo object
-    char *tensorInfoClassName = "ai/onnxruntime/TensorInfo";
-    clazz = (*jniEnv)->FindClass(jniEnv, tensorInfoClassName);
-    jmethodID tensorInfoConstructor = (*jniEnv)->GetMethodID(jniEnv,clazz, "<init>", "([JLai/onnxruntime/OnnxJavaType;Lai/onnxruntime/TensorInfo$OnnxTensorType;)V");
-    //printf("TensorInfo class %p, methodID %p\n",clazz,tensorInfoConstructor);
-    jobject tensorInfo = (*jniEnv)->NewObject(jniEnv, clazz, tensorInfoConstructor, shape, javaDataType, onnxTensorTypeJava);
-    return tensorInfo;
+  // Extract the information from the info struct.
+  ONNXTensorElementDataType onnxType;
+  OrtErrorCode code = checkOrtStatus(jniEnv, api, api->GetTensorElementType(info, &onnxType));
+  if (code != ORT_OK) {
+    return NULL;
+  }
+  size_t numDim = 0;
+  code = checkOrtStatus(jniEnv, api, api->GetDimensionsCount(info, &numDim));
+  if (code != ORT_OK) {
+    return NULL;
+  }
+  //printf("numDim %d\n",numDim);
+  int64_t* dimensions = (int64_t*) malloc(sizeof(int64_t)*numDim);
+  code = checkOrtStatus(jniEnv, api, api->GetDimensions(info, dimensions, numDim));
+  if (code != ORT_OK) {
+    return NULL;
```

Comment:
>return NULL; [](http://example.com/codeflow?start=4&length=12)

dimensions are leaked.
One way to handle this is to declare a fixed array on the stack. Our max shape is actually 5. You can also make it 10, it does not matter for such small things. This way no need to free it.

Reply:
Fixed.